yarn-project/acir-simulator/src/acvm/serialize.ts

```diff
@@ -33,12 +34,16 @@ function adaptBufferSize(originalBuf: Buffer) {
  * @param value - The value to convert.
  * @returns The ACVM field.
  */
-export function toACVMField(value: AztecAddress | EthAddress | Fr | Buffer | boolean | number | bigint): ACVMField {
+export function toACVMField(
+  value: AztecAddress | EthAddress | Fr | Buffer | boolean | number | bigint | ACVMField,
+): ACVMField {
   let buffer;
   if (Buffer.isBuffer(value)) {
     buffer = value;
   } else if (typeof value === 'boolean' || typeof value === 'number' || typeof value === 'bigint') {
     buffer = new Fr(value).toBuffer();
+  } else if (typeof value === 'string') {
+    buffer = Fr.fromString(value).toBuffer();
```

Comment:
Maybe we can add a type guard for isACVMField that checks that it's a hex value of the correct length and if is ACVMField just short circuit the conversion to/from buffer? It'd remove any performance impact of this refactor :D

Reply:
The size and such should already be handled by the `fromString` that was why I just ended up using that 👀. The frombuffer is reading 32 bytes, so if it is of wrong size should break there.

```typescript
  /**
   * Create a Fr instance from a hex-encoded string.
   * The input 'address' can be either prefixed with '0x' or not, and should have exactly 64 hex characters.
   * Throws an error if the input length is invalid or the address value is out of range.
   *
   * @param address - The hex-encoded string representing the field element.
   * @returns A Fr instance.
   */
  static fromString(address: string) {
    return Fr.fromBuffer(Buffer.from(address.replace(/^0x/i, ''), 'hex'));
  }
```

Reply:
Oh yeah I wasn't meaning as validation, It's because, suppose that the toACVMField receives an ACVMField (where it could just return the parameter without any operation)
Then we now would be doing:
Fr.fromString() => internally does Fr.fromBuffer() and Buffer.from(string)
Then fr.toBuffer()
Then buffer.toString('hex')

so that's many conversions to/from buffer when if it's sure about the type it can just return the parameter without doing anything. So with a type guard, you could:

```typescript
function isACVMField(value: any): value is ACVMField {
  ... // Check here if hex string and a hex regex with the ACVMField length for example
}

// Then in toACVMField
...
if (isACVMField(value)) {
    return value
}
```